internal/csi-addons/rbd/replication.go

```diff
@@ -326,7 +326,13 @@ func (rs *ReplicationServer) DisableVolumeReplication(ctx context.Context,
 	case librbd.MirrorImageDisabling:
 		return nil, status.Errorf(codes.Aborted, "%s is in disabling state", volumeID)
 	case librbd.MirrorImageEnabled:
-		return corerbd.DisableVolumeReplication(rbdVol, mirroringInfo, force)
+		err = rbdVol.DisableVolumeReplication(mirroringInfo, force)
+		if err != nil {
+			return nil, err
```

Comment:
this should not return `err` directly, but a gRPC error with a suitable code. `DisableVolumeReplication()` did not seem to have done that in all cases, that was not correct. With this refactoring it becomes clear, and you should address that.